src/main/scala/cromwell/binding/WdlBinding.scala

```diff
@@ -115,6 +114,11 @@ object WdlBinding {
     }
   }
 
+  /* All MemberAccess ASTs that are not contained in other MemberAccess ASTs */
+  def findTopLevelMemberAccesses(expr: AstNode) = WdlBinding.findAstsWithTrail(expr, "MemberAccess").filter {
+    case (k, v) => !(v map {case a:Ast => a.getName} contains "MemberAccess")
```

Comment:
I'll try to write a for comprehension.  When you say "multiple cased higher order functions in a line", what do you mean?  I also don't know if I agree with "start breaking things down into funcs".  The function in this case would be a one liner, something to the effect of `filterOutAstPathsWithAncestor` or even worse `filterOutAstPathsWithMemberAccessAncestor` if you only wanted it to take one parameter.  I never like the presence of those one-line functions that are only ever used in one place... I say that if this code is used in more than once place, why not refactor it out then?